internal/commands/build/build.go

```diff
@@ -128,37 +133,66 @@ func runBuild(dockerCli command.Cli, contextPath string, opt buildOptions) error
 	}
 	defer app.Cleanup()
 
-	bundle, err := buildImageUsingBuildx(app, contextPath, opt, dockerCli)
+	bndl, err := buildImageUsingBuildx(app, contextPath, opt, dockerCli)
 	if err != nil {
 		return err
 	}
 
-	var ref reference.Reference
-	ref, err = packager.GetNamedTagged(opt.tag)
+	out, err := getOutputFile(dockerCli.Out(), opt.quiet)
 	if err != nil {
 		return err
 	}
 
-	id, err := packager.PersistInBundleStore(ref, bundle)
+	id, err := persistTags(bndl, opt.tags, opt.imageIDFile, out, dockerCli.Err())
 	if err != nil {
 		return err
 	}
 
-	if opt.imageIDFile != "" {
-		if err = ioutil.WriteFile(opt.imageIDFile, []byte(id.Digest().String()), 0644); err != nil {
-			fmt.Fprintf(dockerCli.Err(), "Failed to write App image ID in %s: %s", opt.imageIDFile, err)
-		}
+	if opt.quiet {
+		_, err = fmt.Fprintln(dockerCli.Out(), id.Digest().String())
 	}
+	return err
+}
 
-	if opt.quiet {
-		fmt.Fprintln(dockerCli.Out(), id.Digest().String())
-		return err
+func persistTags(bndl *bundle.Bundle, tags cliOpts.ListOpts, iidFile string, outWriter io.Writer, errWriter io.Writer) (reference.Digested, error) {
+	var id reference.Digested
+	var onceWriteIIDFile sync.Once
+	if tags.Len() == 0 {
+		err := tags.Set("")
+		if err != nil {
+			return nil, err
+		}
 	}
-	fmt.Fprintf(dockerCli.Out(), "Successfully built %s\n", id.String())
-	if ref != nil {
-		fmt.Fprintf(dockerCli.Out(), "Successfully tagged %s\n", ref.String())
+	for _, tag := range tags.GetAll() {
+		ref, err := packager.GetNamedTagged(tag)
+		if err != nil {
+			return nil, err
+		}
+		id, err = persistInBundleStore(&onceWriteIIDFile, outWriter, errWriter, bndl, ref, iidFile)
+		if err != nil {
+			return nil, err
+		}
+		if tag != "" {
+			fmt.Fprintf(outWriter, "Successfully tagged app %s\n", ref.String())
+		}
 	}
-	return err
+	return id, nil
+}
+
+func persistInBundleStore(once *sync.Once, out io.Writer, errWriter io.Writer, b *bundle.Bundle, ref reference.Reference, iidFileName string) (reference.Digested, error) {
+	id, err := packager.PersistInBundleStore(ref, b)
+	if err != nil {
+		return nil, err
+	}
+	once.Do(func() {
+		fmt.Fprintf(out, "Successfully built app %s\n", id.String())
```

Comment:
The text changed, is that ok?

Reply:
This is something I would like to discuss.
The SHA that this message shows is not an image, but a bundle. This is exactly the same message shown by the "normal builder" and `docker-compose` to show the SHA of a `docker image`.
IMO, that's confusing to have this message building another thing and saying the same.

Reply:
Well, actually it's an image... Not a container image but still :)
Any opinion @ndeloof as you first wrote this log I guess?

Reply:
"app image" is the recommended name for this (se naming doc)

Reply:
Reason to show status this way is to "mimic the docker CLI" as our _UX mantra_ :P

Reply:
hmmm... Looks like now we have 3 alternatives:
- "Successfully built"
- "Successfully built app"
- "Successfully built app image"

What do you think @chris-crone @eunomie @zappy-shu ?

Reply:
With BuildKit the final line is:
```
=> => writing image sha256:6f71466d421e3ab378cf91c03bfd4d00ecfc1ba7dc3a37ccb69c02b2ef0a2f6b
```

Without, it's:
```
Successfully built 965ea09ff2eb
```

I would say that we should use option 3 to make sure that it's clear.